eis_toolkit/cli.py

```diff
@@ -1179,7 +1460,137 @@ def winsorize_transform_cli(
+):
+    """Calculate area under curve (AUC)."""
+    from eis_toolkit.validation.calculate_auc import calculate_auc
+
+    typer.echo("Progress: 10%")
+
+    with rasterio.open(x_values) as x_src:
+        x_array = x_src.read(1)
+        typer.echo("Progress: 25%")
+
+    with rasterio.open(y_values) as y_src:
+        y_array = y_src.read(1)
+        typer.echo("Progress: 50%")
+
+    auc_value = calculate_auc(x_array, y_array)
+    typer.echo(f"AUC Value: {auc_value}")
+    typer.echo("Progress: 75%")
+
+    # Save the AUC value to the output file
+    with open(output_file, 'w') as output:
+        output.write(f"AUC Value: {auc_value}")
+    typer.echo("Progress: 100%")
+    typer.echo(f"Calculate area under curve complete, writing raster to {output_file}.")
+
+
+# CALCULATE BASE METRICS
+@app.command()
+def calculate_base_metrics_cli(
+   input_raster: Annotated[Path, INPUT_FILE_OPTION],
+   input_deposits: Annotated[Path, INPUT_FILE_OPTION],
+   input_negatives: Annotated[Path, INPUT_FILE_OPTION],
+   output_vector: Annotated[Path, OUTPUT_FILE_OPTION],
+   band: int = 1,
+):
+    """Calculate true positive rate, proportion of area and false positive rate values for different thresholds."""
+    from eis_toolkit.validation.calculate_base_metrics import calculate_base_metrics
+
+    typer.echo("Progress: 10%")
+
+    with rasterio.open(input_raster) as src:
+        raster = src.read()
+        typer.echo("Progress: 20%")
+
+    deposits = gpd.read_file(input_deposits)
+    typer.echo("Progress: 40%")
+
+    negatives = gpd.read_file(input_negatives)
+    typer.echo("Progress: 60%")
+
+    metrics = calculate_base_metrics(
+        raster=raster,
+        deposits=deposits,
+        band=band,
+        negatives=negatives,
+    )
+    typer.echo("Progress: 80%")
+
+    metrics.to_csv(output_vector)
+    typer.echo("Progress: 100%")
+    typer.echo(f"Calculate base metrics complete, writing vector to {output_vector}.")
+
+
+# PLOT CORRELATION MATRIX
+# TODO! This will need some work.
+@app.command()
+def correlation_matrix_cli(
+    input_vector: Annotated[Path, INPUT_FILE_OPTION],
+    annotate: bool,
+    cmap: str,
+    plot_title: str,
+    output_png: Annotated[Path, OUTPUT_FILE_OPTION]
+):
+    """Create a Seaborn heatmap to visualize correlation matrix."""
+    from eis_toolkit.validation.plot_correlation_matrix import plot_correlation_matrix
+
+    typer.echo("Progress: 10%")
+
+    geodataframe = gpd.read_file(input_vector)
+    dataframe = pd.DataFrame(geodataframe.drop(columns="geometry"))
+    typer.echo("Progress: 35%")
+
+    result_ax = plot_correlation_matrix(
+        matrix=dataframe,
+        annotate=annotate,
+        cmap=cmap,
+        plot_title=plot_title,
+        kwargs=None)
+    typer.echo("Progress: 70%")
+
+    result_ax.figure.savefig(output_png, format='png', bbox_inches='tight')
+    typer.echo("Progress: 100%")
+    typer.echo(f"Plot correlation matrix complete, writing image to {output_png}.")
```

Comment:
Just make the "bare" parameters as typer.Option(). I am not entirely sure if the plugin will call this function, though. As with the other plots in EDA, all the required libraries of this function are available in QGIS and displaying plots directly in QGIS can be beneficial. We can still keep this implementation here.